custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -1164,3 +1172,12 @@ def create_custom_inference_model_folder(code_dir, output_dir):
         fp.write(readme)
     if files_in_output & copied_files:
         print("Files were overwritten: {}".format(files_in_output & copied_files))
+
+
+def get_default_parameter_values(model_metadata: Dict) -> Dict[str, Union[int, float, str]]:
+    hyper_param_config = model_metadata.get(ModelMetadataKeys.HYPERPARAMETERS, {})
+    default_params = {}
+    for param_config in hyper_param_config:
+        if param_config.get("default"):
+            default_params[param_config["name"]] = param_config["default"]
```

Comment:
We need to get default values even if they do not supply a default value. It is defined in DR as:
- int/float: the min value
- select: the first value in their array
- string: empty string
- multi: the first multi type's default value

Reply:
> int/float: the min value

Not sure if this is the common practice to use e.g., `Integer.MIN_VALUE` as the default value if int/float is not provided...

> select: the first value in their array
If there is no `hyperparameters` section in model-metadata.yaml or no model-metadata.yaml is provided, we probably don't know what is the allowed values in list?

Reply:
sorry for not being more clear, min value being the min value the user provides (since it's a required filed).

If there is no hyperparameters at all in model-metadata, then we don't need to worry about providing default values at all at runtime.

Reply:
Here's a more explicit example. User does not provide any defaults to their parameters:
```
hyperparameters:
  - name: penalty  # default will be l1
    type: select
    values:
      - l1
      - l2
      - elasticnet
  - name: dual  # default will be 0
    type: int
    min: 0
    max: 1
  - name: tol  # default will be 0
    type: float
    min: 0
    max: 1
  - name: solver  # default will be empty string
    type: string
```

Reply:
The logic was updated. 
https://github.com/datarobot/datarobot-user-models/blob/c40e34a2ece79f88bc649aa8235fc6935c0d8788/custom_model_runner/datarobot_drum/drum/drum.py#L1177-L1186